lib/PersistedPaneset/PersistedPaneset.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useLocalStorage, writeStorage } from '@rehooks/local-storage';
+
+import { Paneset } from '@folio/stripes-components';
+
+const propTypes = {
+  appId: PropTypes.string.isRequired,
+  children: PropTypes.node,
+  id: PropTypes.string.isRequired,
+};
+
+const PersistedPaneset = ({
+  appId,
+  children,
+  id: panesetId,
+}) => {
+  const persistKey = `${appId}/${panesetId}/layout`;
+  const [storedPanesetLayout] = useLocalStorage(persistKey);
+
+  return (
+    <Paneset
+      id={panesetId}
+      initialLayouts={storedPanesetLayout}
+      onResize={({ layoutCache }) => {
+        // We don't want to persist any layout that contains a NaN so filter those out first.
+        const persistedLayoutCache = layoutCache.filter(layout => {
+          const widths = Object.values(layout);
+          const hasNaN = widths.some(width => width.includes('NaN'));
```

Comment:
This is here because I noticed that occasionally - especially during initial mounting and page load - some layouts resolve a Pane to a width of `"NaN%"`. This generally occurs when rendering a Paneset with a single Pane in it, such as when filters are hidden and no record is selected.